lib/iris/tests/iris_test_pytest.py

```diff
@@ -0,0 +1,1033 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Provides testing capabilities and customisations specific to Iris.
+
+.. note:: This module needs to control the matplotlib backend, so it
+          **must** be imported before ``matplotlib.pyplot``.
+
+The primary class for this module is :class:`IrisTest`.
+
+"""
+
+import collections
+from collections.abc import Mapping
+import contextlib
+import difflib
+import filecmp
+import functools
+import gzip
+import inspect
+import io
+import json
+import math
+import os
+import os.path
+from pathlib import Path
+import re
+import shutil
+import subprocess
+import sys
+from typing import AnyStr
+
+# from unittest import mock
+import warnings
+import xml.dom.minidom
+import zlib
+
+import numpy as np
+import numpy.ma as ma
+import pytest
+import requests
+
+import iris.config
+import iris.cube
+import iris.fileformats
+import iris.tests.graphics as graphics
+import iris.util
+
+MPL_AVAILABLE = graphics.MPL_AVAILABLE
+
+
+try:
+    from osgeo import gdal  # noqa
+except ImportError:
+    GDAL_AVAILABLE = False
+else:
+    GDAL_AVAILABLE = True
+
+try:
+    import iris_sample_data  # noqa
+except ImportError:
+    SAMPLE_DATA_AVAILABLE = False
+else:
+    SAMPLE_DATA_AVAILABLE = True
+
+try:
+    import nc_time_axis  # noqa
+
+    NC_TIME_AXIS_AVAILABLE = True
+except ImportError:
+    NC_TIME_AXIS_AVAILABLE = False
+
+try:
+    # Added a timeout to stop the call to requests.get hanging when running
+    # on a platform which has restricted/no internet access.
+    requests.get("https://github.com/SciTools/iris", timeout=10.0)
+    INET_AVAILABLE = True
+except requests.exceptions.ConnectionError:
+    INET_AVAILABLE = False
+
+try:
+    import stratify  # noqa
+
+    STRATIFY_AVAILABLE = True
+except ImportError:
+    STRATIFY_AVAILABLE = False
+
+#: Basepath for test results.
+_RESULT_PATH = os.path.join(os.path.dirname(__file__), "results")
+
+if "--data-files-used" in sys.argv:
+    sys.argv.remove("--data-files-used")
+    fname = "/var/tmp/all_iris_test_resource_paths.txt"
+    print("saving list of files used by tests to %s" % fname)
+    _EXPORT_DATAPATHS_FILE = open(fname, "w")
+else:
+    _EXPORT_DATAPATHS_FILE = None
+
+
+if "--create-missing" in sys.argv:
+    sys.argv.remove("--create-missing")
+    print("Allowing creation of missing test results.")
+    os.environ["IRIS_TEST_CREATE_MISSING"] = "true"
+
+
+def main():
```

Comment:
This hasn't been working for a while. So no point in transferring over, but we do need to investigate, so an issue should be raised.

This statement also applies to the arguments referenced e.g. `--data-files-used`

Reply:
Turns out it IS working, I had just misunderstood it. Anyway please remove all that stuff from this file.

Here is the issue: #5809